packages/preview-middleware-client/src/adp/api-handler.ts

```diff
@@ -171,7 +171,10 @@ export async function getDataSourceAnnotationFileMap(): Promise<AnnotationDataSo
  * @returns {CodeExtResponse} Returns path to existing controller if found
  */
 export async function getExistingController(controllerName: string): Promise<CodeExtResponse> {
-    return request<CodeExtResponse>(`${ApiEndpoints.CODE_EXT}/${controllerName}` as ApiEndpoints, RequestMethod.GET);
+    return request<CodeExtResponse>(
+        `${ApiEndpoints.CODE_EXT}?name=${controllerName}` as ApiEndpoints,
```

Comment:
Query parameters should be encoded using `encodeURIComponent`.

Reply:
Changed in [483af3b](https://github.com/SAP/open-ux-tools/pull/3206/commits/483af3b55a3e0e69f46f6e89a9f603ded3dafeca). Decided to use URLSearchParams, which is more robust if in the future we will wanna add more query parameters.